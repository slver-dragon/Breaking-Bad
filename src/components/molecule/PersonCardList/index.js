import React from "react";
import style from "./style.module.scss";
import { Link } from "react-router-dom";

let statusStyle = "";

const PersonCard = ({ name, birthday, img, status, id }) => {
  switch (status) {
    case "Alive": {
      statusStyle = style.alive;
      break;
    }
    case "Deceased": {
      statusStyle = style.deceased;
      break;
    }
    default: {
      statusStyle = style.presumedDead;
    }
  }

  return (
    <div className={style.container}>
      <Link to={`/person/${id}`} className={style.main}>
        <div className={style.photoWrapper}>
          <img className={style.photo} src={img} alt={name} />
        </div>
        <div className={style.inform}>
          <p className={style.state + " " + statusStyle}>{status}</p>
          <span className={style.name}>{name}</span>
          <span className={style.birthday}>{birthday} </span>
        </div>
      </Link>
    </div>
  );
};

export default PersonCard;
