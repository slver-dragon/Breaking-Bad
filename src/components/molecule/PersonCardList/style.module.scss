@import "../../../assets/styles/style";

.container {
  border-top: 0.0625rem solid $gray;
  .main {
    display: flex;
    justify-content: space-between;
    @include standarFont;
    margin: 0.75rem 0;
    text-decoration: none;
    .photoWrapper {
      max-width: 4rem;
      max-height: 4rem;
      object-fit: cover;
      margin-right: 0.2rem;
      overflow: hidden;
      .photo {
        max-width: 4rem;
        height: auto;
      }
    }
    .inform {
      display: inline-grid;
      grid-template-columns: 1fr 1fr 1fr;
      align-items: center;
      justify-content: space-around;
      flex: 1 1 auto;
      @include mediumFont;
      .state {
        display: inline-block;
        padding: 0.5rem 0.75rem;
        margin: 1rem 0 0.75rem;
        justify-self: center;
      }
      .alive {
        @include stateOnColor;
      }
      .deceased {
        @include stateOffColor;
      }
      .presumedDead {
        @include statePDColor;
      }
      .name {
        color: $black;
        justify-self: center;
      }
      .birthday {
        color: $gray;
        justify-self: center;
      }
    }
  }
}
