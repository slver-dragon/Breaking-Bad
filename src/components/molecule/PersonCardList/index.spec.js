import React from "react";
import { shallow } from "enzyme";
import PersonCardList from ".";

describe("Testing molecule Paginator:", () => {
  let component = null;
  const setComponent = (props) => shallow(<PersonCardList {...props} />);

  test("is rendering", () => {
    component = setComponent();
    expect(component).toMatchSnapshot();
  });

  test("status change", () => {
    component = setComponent({ status: "Alive" });
    expect(component.find(".alive").length).toBe(1);
    component = setComponent({ status: "Deceased" });
    expect(component.find(".deceased").length).toBe(1);
    component = setComponent({ status: "Presumed Dead" });
    expect(component.find(".presumedDead").length).toBe(1);
  });
});