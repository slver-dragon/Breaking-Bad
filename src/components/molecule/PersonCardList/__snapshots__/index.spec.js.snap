// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing molecule Paginator: is rendering 1`] = `
<div
  className="container"
>
  <Link
    className="main"
    to="/person/undefined"
  >
    <div
      className="photoWrapper"
    >
      <img
        className="photo"
      />
    </div>
    <div
      className="inform"
    >
      <p
        className="state presumedDead"
      />
      <span
        className="name"
      />
      <span
        className="birthday"
      >
         
      </span>
    </div>
  </Link>
</div>
`;
